rest/oidc_api.go

```diff
@@ -15,34 +15,38 @@ import (
 	"net/url"
 	"time"
 
-	"github.com/coreos/go-oidc/oauth2"
 	"github.com/couchbase/sync_gateway/auth"
 	"github.com/couchbase/sync_gateway/base"
+	"golang.org/x/oauth2"
 )
 
 const (
-	OIDC_AUTH_RESPONSE_TYPE = "response_type"
-	OIDC_AUTH_CLIENT_ID     = "client_id"
-	OIDC_AUTH_SCOPE         = "scope"
-	OIDC_AUTH_REDIRECT_URI  = "redirect_uri"
-	OIDC_AUTH_STATE         = "state"
-
-	OIDC_RESPONSE_TYPE_CODE     = "code"
-	OIDC_RESPONSE_TYPE_IMPLICIT = "id_token%20token"
+	requestParamCode         = "code"
+	requestParamProvider     = "provider"
+	requestParamOffline      = "offline"
+	requestParamError        = "error"
+	requestParamErrorDesc    = "error_description"
+	requestParamRefreshToken = "refresh_token"
+	requestParamState        = "state"
+	requestParamScope        = "scope"
+	requestParamRedirectURI  = "redirect_uri"
+)
+
+const (
+	keyIDToken = "id_token"
 )
 
 type OIDCTokenResponse struct {
-	IDToken      string `json:"id_token"`                // ID token, from OP
-	RefreshToken string `json:"refresh_token,omitempty"` // Refresh token, from OP
-	SessionID    string `json:"session_id,omitempty"`    // Sync Gateway session ID
-	Username     string `json:"name,omitempty"`          // Sync Gateway user name
-	AccessToken  string `json:"access_token,omitempty"`  // Access token, from OP
-	TokenType    string `json:"token_type,omitempty"`    // Access token type, from OP
-	Expires      int    `json:"expires_in,omitempty"`    // Access token expiry, from OP
+	IDToken      string    `json:"id_token"`                // ID token, from OP
+	RefreshToken string    `json:"refresh_token,omitempty"` // Refresh token, from OP
+	SessionID    string    `json:"session_id,omitempty"`    // Sync Gateway session ID
+	Username     string    `json:"name,omitempty"`          // Sync Gateway user name
+	AccessToken  string    `json:"access_token,omitempty"`  // Access token, from OP
+	TokenType    string    `json:"token_type,omitempty"`    // Access token type, from OP
+	Expires      time.Time `json:"expires_in,omitempty"`    // Access token expiry, from OP
```

Comment:
This doesn't align with the OIDC spec (and would break backwards compatibility even if it was to spec):
```
expires_in
OPTIONAL. Expiration time of the Access Token in seconds since the response was generated.
```

Reply:
Added changes to keep the expiration time of the access token in seconds since the response was generated. Now it is aligned with the OpenID Connect spec and is backward compatible.